### PR TITLE
BOAC-1354, added '/api/curated_cohorts/my/<sid>' endpoint; checkbox fixes per Angular upgrade

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1354
https://jira.ets.berkeley.edu/jira/browse/BOAC-1131

And, the front-end cohorts in rootScope are refreshed with server-side response of `/add/student`, etc.

### CODE REVIEW

File: boac/api/curated_cohort_controller.py [hunk 1]
```diff
@@ -105,6 +105,12 @@ def get_students_with_alerts(cohort_id):
     return tolerant_jsonify(sort_students_by_name(students_with_alerts))
 
 
+@app.route('/api/curated_cohorts/my/<sid>')
+@login_required
+def curated_cohort_ids_per_sid(sid):
+    return tolerant_jsonify(CuratedCohort.curated_cohort_ids_per_sid(user_id=current_user.id, sid=sid))
```
Comment (johncrossman) [inline boac/api/curated_cohort_controller.py]: Intentionally lightweight. Front-end only needs `ids`.

File: boac/static/app/cohort/curated/selectorComponent.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+  angular.module('boac').component('curatedCohortSelector', {
+    bindings: {
+      students: '='
+    },
+    controller: CuratedCohortSelector,
+    templateUrl: '/static/app/cohort/curated/selector.html'
+  });
+
+}(window.angular));
```
Comment (johncrossman) [inline boac/static/app/cohort/curated/selectorComponent.js]: This was supposed to be a `git move selectorDirective.js` (see below). Can't win 'em all!

File: boac/static/app/cohort/curated/cohortController.js [hunk 3]
```diff
@@ -84,20 +85,18 @@
...
 
     $scope.removeFromCuratedCohort = function(student) {
-      curatedCohortFactory.removeStudent($scope.cohort, student).then(function() {
-        $scope.cohort.students = _.remove($scope.cohort.students, function(s) {
-          return s.sid !== student.sid;
-        });
+      $scope.students = _.remove($scope.students, function(s) {
+        return s.sid !== student.sid;
       });
+      curatedCohortFactory.removeStudent($scope.cohortId, student.sid);
```
Comment (lyttam) [inline boac/static/app/cohort/curated/cohortController.js]: @johncrossman if this request fails then `$scope.students` will be out of sync with the backend, is that ok?

Comment (johncrossman) [reply boac/static/app/cohort/curated/cohortController.js]: A good point. I'll amend. Thanks @lyttam

**Comment by paulkerschen** [comment]
Components!!

**Pull request merged**